src/Store/saga/index.ts

```diff
@@ -24,10 +26,34 @@ export default function* saga(): SagaIterator {
   yield fork(interpret);
   yield fork(pools);
 
+  yield takeEvery(stats.matchDispatch, handleFetchStatistics);
   yield takeEvery(signOut.type, handleSignOut);
   yield takeEvery(showNotification.type, autoHideNotification);
 }
 
+/**
+ * Special saga for the statistics route to inject the guildId into the action.
+ * Should probably be generalized.
+ * @param action - the satistics respones action dispatched upon receiving a response from the api
+ */
+function* handleFetchStatistics(
```

Comment:
I didn't think of this before, but you might not actually need the additional action/saga. `RestSuccess` should include the `routeArgs` which should have the guild id available for you.

Reply:
nice brain